### ISSUE TITLE
Improve unit tests for merging observables in event service

### ISSUE THREAD

**Issue by Platonn** [SAP/spartacus#7630]
In PR #7582 only one complex unit test was added for merging observable. However it's not super readable and can be improved. Moreover it would be nice to have some unit tests for checking the internal subscribtions - whether they are disposed correctly

- internal subscriptions (can be checked by passing fake new Observables with callback when they are subscribed and unsubscribed from)

- register(add):
  - check it's added/removed to/from array
  -  check when it already have been added
  - check it's not subscribed twice

- multicasting:
  -  add two subscriptions and check that each source (i.e. 2) are subscribed only once
  - unsubscribe 1 and check that each source is still subscribed
  - unsubscribe 2 and check that all sources are unsubscribed from

- after unregister(remove):
  -  check internal state that subscriptions are removed
  - check internal state that subscriptions are not active anymore
  - check it's removed from array

- after unsubscribing all consumers:
  -  check internal state that subscriptions are removed
  -  check internal state that subscriptions are not active anymore (subscription.active === false or sth. like that)
  - check sources are still in the array

**Comment by Platonn** [comment]
Snippet how to spy on subcriptions and unsubscriptions of sourceA and sourceB
```typescript
    const originalSubscribeA = sourceA.subscribe.bind(sourceA);
    const originalSubscribeB = sourceB.subscribe.bind(sourceA);

    const onUnsubscribeA = jasmine.createSpy('unsubscribed from source A');
    const onUnsubscribeB = jasmine.createSpy('unsubscribed from source B');

    spyOn(sourceA, 'subscribe').and.callFake((...args) => {
      const sub = originalSubscribeA(...args);
      const originalUnsubscribe = sub.unsubscribe.bind(sub);
      spyOn(sub, 'unsubscribe').and.callFake(() => {
        onUnsubscribeA();
        return originalUnsubscribe();
      });
      return sub;
    });

    spyOn(sourceB, 'subscribe').and.callFake((...args) => {
      const sub = originalSubscribeB(...args);
      const originalUnsubscribe = sub.unsubscribe.bind(sub);
      spyOn(sub, 'unsubscribe').and.callFake(() => {
        onUnsubscribeB();
        return originalUnsubscribe();
      });
      return sub;
    });
```
